src/GitHub.App/Factories/ViewViewModelFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Windows;
+using GitHub.Exports;
+using GitHub.Services;
+using GitHub.ViewModels;
+
+namespace GitHub.Factories
+{
+    /// <summary>
+    /// Factory for creating views and view models.
+    /// </summary>
+    [Export(typeof(IViewViewModelFactory))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class ViewViewModelFactory : IViewViewModelFactory
+    {
+        readonly IGitHubServiceProvider serviceProvider;
+
+        [ImportingConstructor]
+        public ViewViewModelFactory(
+            IGitHubServiceProvider serviceProvider,
+            ICompositionService cc)
+        {
+            this.serviceProvider = serviceProvider;
+            cc.SatisfyImportsOnce(this);
+        }
+
+        [ImportMany(AllowRecomposition = true)]
```

Comment:
Visual Studio's MEF implementation is static (the graph is created once and cached). New exports/assemblies can't be added to the catalog on the fly. Having `AllowRecomposition = true` is confusing because recomposition is impossible.

Reply:
Ok, well I'm fine with removing it! We should probably do it in a new PR though in case it causes problems.